python/lsst/ci/build.py

```diff
@@ -328,15 +332,59 @@ def write_status(self):
             yaml.dump(status, sf, encoding="utf-8", default_flow_style=False)
 
     @staticmethod
-    def run(args):
+    def get_agent() -> str:
+        """
+        Determine the system agent based on OS type and architecture.
+
+        Returns
+        -------
+        str
+            The formatted agent string, such as "apple_arm64" or
+            "linux_x86_64". If the system platform or architecture
+            cannot be determined, returns "unknown_agent".
+        """
+        os_type = sys.platform
+        arch = platform.machine()
+
+        if not os_type or not arch:
+            agent = "unknown_agent"
+        else:
+            match os_type:
+                case "darwin":
+                    agent = f"apple_{arch}"
```

Comment:
Is this our mapping or does it need to match someone else's?  Why `apple` specifically and not `macos` or `osx`?